### DISCUSSION TITLE
_sw_handle_request() takes 3 positional arguments but 4 were given

### DISCUSSION THREAD

**Discussion by suntangi** [apache/skywalking#11508] [Q&A]
I see this exception when using apache-skywalking==1.0.1, wslink==1.6.4

```
  File "/usr/local/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 509, in start
    self._handle_request(request, start, request_handler)
TypeError: _sw_handle_request() takes 3 positional arguments but 4 were given
Traceback (most recent call last):
ERROR:aiohttp.server:Unhandled exception
```

see code in skywalking:

> https://github.com/apache/skywalking-python/blob/master/skywalking/plugins/sw_aiohttp.py

line 83
`  async def _sw_handle_request(self, request: BaseRequest, start_time: float):`
line 122
`    RequestHandler._handle_request = _sw_handle_request`

see code in aiohttp:

> [https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_protocol.py](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_protocol.py)
line 458

```
  async def _handle_request(
        self,
        request: BaseRequest,
        start_time: float,
        request_handler: Callable[[BaseRequest], Awaitable[StreamResponse]],
    ) -> Tuple[StreamResponse, bool]:
```
line 509

```
                # a new task is used for copy context vars (#3406)
                task = self._loop.create_task(
                    self._handle_request(request, start, request_handler)
                )
```

seems skywalking try to replace _handle_request of aiohttp, but has mismatched args.
this exception only occurs while using wslink 

please tell me what to do.

**Comment by wu-sheng** [comment]
Friendly ping @Superskyyy @kezhenxu94

**Comment by Superskyyy** [comment]
@suntangi  Hi, this must be due to the version incompatibility of plugin and recent versions of AIOHTTP that can be easily fixed. Could you kindly share your aiohttp version? Is your version currently covered by the test matrix? https://skywalking.apache.org/docs/skywalking-python/next/en/setup/plugins/.

The thing to modify is to add the additional `request_handler` field to the method args. 

I just checked the code blame for aiohttp, it seems the original 2 arg method version is quite long ago around aiohttp v3.7.0, so lets just replace with the new args version and remove support for such old versions. 



If you would like to contribute a fix please go ahead and do so, otherwise let me know and I will get that fixed within few days.


```python
support_matrix -> Update to test 3.8.x and 3.9.x
Line83
async def _sw_handle_request(
	self, request: BaseRequest, 
	start_time: float, 
	request_handler: Callable[[BaseRequest], Awaitable[StreamResponse]],
):

```

**Comment by suntangi** [comment]
@Superskyyy 

Thanks for your reply!

aiohttp==3.8.4,  not covered by the test matrix

In my case, I simply disable aiohttp plugin to avoid exceptions during the main process, not planning to do more research. Please schedule your time to fix it.

`config.agent_disable_plugins = ['sw_aiohttp']`

**Comment by Superskyyy** [comment]
> @Superskyyy
> 
> Thanks for your reply!
> 
> aiohttp==3.8.4, not covered by the test matrix
> 
> In my case, I simply disable aiohttp plugin to avoid exceptions during the main process, not planning to do more research. Please schedule your time to fix it.
> 
> `config.agent_disable_plugins = ['sw_aiohttp']`

No problem, thanks for the report anyway!

**Reply by XXMY** [reply]
> > @Superskyyy
> > Thanks for your reply!
> > aiohttp==3.8.4, not covered by the test matrix
> > In my case, I simply disable aiohttp plugin to avoid exceptions during the main process, not planning to do more research. Please schedule your time to fix it.
> > `config.agent_disable_plugins = ['sw_aiohttp']`
> 
> No problem, thanks for the report anyway!

How about this issue, did you fixed?
